src/aframe_components/custom-components.ts

```diff
@@ -20,24 +25,100 @@ interface State {
 export interface Guidance extends Component {
   state: State;
   pointPositions: Vector3[];
+  kdtree: KDNode | null;
+}
+
+export interface KDNode {
+  position: Vector3;
+  left: KDNode | null;
+  right: KDNode | null;
+}
```

Comment:
Not a request for a change, just curious if you gave any consideration to implementing this as a class instead of an interface with associated functions.

Reply:
I was under the impression that using an interface was a bit lighter on overhead, and so more efficient, but that a class organization may be helpful for readability. Would you recommend one over the other?